src/Illuminate/Foundation/Testing/DatabaseConnectionFactory.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+namespace Illuminate\Foundation\Testing;
+
+use Illuminate\Contracts\Container\Container;
+use Illuminate\Contracts\Database\Connectors\ConnectionFactory as ConnectionFactoryContract;
+use Illuminate\Database\Connectors\ConnectionFactory;
+
+/**
+ * @internal
+ */
+class DatabaseConnectionFactory extends ConnectionFactory
+{
+    /**
+     * List of cached database connections.
+     *
+     * @var array<string, \Illuminate\Database\Connection>
+     */
+    protected static array $cachedConnections = [];
+
+    /**
+     * Create a new connection factory instance.
+     *
+     * @param  \Illuminate\Contracts\Container\Container  $container
+     * @return void
+     */
+    public function __construct(
+        Container $container,
+        protected ConnectionFactoryContract $factory
+    ) {
+        parent::__construct($container);
+    }
+
+    /**
+     * Establish a PDO connection based on the configuration.
+     *
+     * @param  array  $config
+     * @param  string  $name
+     * @return \Illuminate\Database\Connection
+     */
+    #[\Override]
+    public function make(array $config, $name)
+    {
+        $key = $name ?? $config['name'];
+
+        // SQLite doesn't have any max connections limitation so it should be safe to just create a new connection between tests.
```

Comment:
*IMO:*
```suggestion
        // In-Memory Databases doesn't have any max connections limitation so it should be safe to just create a new connection between tests.
        // Because some tests may be depend on thier volatile, we should always create new connections to avoid carrying over previous data.
```

In addition to the current comment,
How about adding that it supports testing assuming the volatility of the In Memory Database?

I initially tried turning this off, thinking it would be better to have consistent behavior regardless of the driver, but many of the tests for the conditions mentioned above failed. So, I understand that this process is necessary from a different perspective than limiting the number of connections.

Reply:
The affect of this behaviour is a huge breaking change for `:memory:` and should only occur with `RefreshDatabase` new improvements for L11.

Reply:
See https://github.com/laravel/framework/pull/47912

Reply:
I caught up, thank you.

In addition to the general case you mention, there were some parts of Laravel's own testing that were related to this. for example:

https://github.com/laravel/framework/blob/41c572ef01841be07304b87c3926a652bddc1d59/tests/Integration/Database/DatabaseSchemaBlueprintTest.php#L35-L38
(This test fails with `table "users" already exists` if we persist the connection.)

Either way, the meaning of the comment is essentially the same, so either is fine. Thank you for confirmation!